Google.PowerShell/PubSub/GcpsTopic.cs

```diff
@@ -0,0 +1,234 @@
+﻿using Google.Apis.Pubsub.v1;
+using Google.Apis.Pubsub.v1.Data;
+using Google.PowerShell.Common;
+using System;
+using System.Management.Automation;
+using System.Net;
+
+namespace Google.PowerShell.PubSub
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Creates new Google Cloud PubSub topics.
+    /// </para>
+    /// <para type="description">
+    /// Creates one or more Gooogle Cloud PubSub topics. Will throw errors if the topics already exist.
+    /// The cmdlet will create the topics in the default project if -Project is not used.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> New-GcpsTopic -Topic "my-topic"</code>
+    ///   <para>This command creates a new topic called "my-topic" in the default project.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> New-GcpsTopic -Topic "topic1", "topic2" -Project "my-project"</code>
+    ///   <para>
+    ///   This command creates 2 topics ("topic1" and "topic2") in the project "my-project".
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/pubsub/docs/publisher#create-a-topic)">
+    /// [Creating a Topic]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "GcpsTopic")]
+    public class NewGcpsTopic : GcpsCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The project to create the topics in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The names of the topics to be created. Topics must not have already existed.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0)]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string[] Topic { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            foreach (string topicName in Topic)
+            {
+                string formattedTopicname = PrefixProjectToTopic(topicName, Project);
+                Topic topic = new Topic() { Name = formattedTopicname };
+                ProjectsResource.TopicsResource.CreateRequest request = Service.Projects.Topics.Create(topic, formattedTopicname);
+                try
+                {
+                    Topic response = request.Execute();
+                    WriteObject(response);
+                }
+                catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.Conflict)
+                {
+                    ErrorRecord errorRecord = new ErrorRecord(
```

Comment:
This is pretty inconsistent with the way we have been doing things in the past. (Either letting the exception bubble up, or just calling `WriteError` with a string.)

Should we be trying to clean up other places in the code to use this more elaborate pattern? Should we try to simplify it some how? e.g. `WriteResourceExistsError("topic", topicName, "TopicAlreadyExists");`?